src/app/components/Arbitrage/Arbitrage.tsx

```diff
@@ -1,77 +1,120 @@
+      amount: number;
+      from: string;
+      to: string;
+      rate: number;
+      earn: number;
+    };
+  };
+}
+
 export function Arbitrage() {
   const { t } = useTranslation();
-  const api = backendUrl[currentChainId];
-  const [show, setShow] = useState(false);
-  const [data, setData] = useState({
-    USDT: {
-      oracleRate: '0',
-      negativeDelta: false,
-      rateToBalance: { amount: 0, from: '', to: '', rate: 0, earn: 0 },
-    },
-  });
+  const { assetRates } = useSelector(selectWalletProvider);
 
-  useEffect(() => {
-    axios
-      .get(api + '/amm/arbitrage')
-      .then(res => setData(res.data))
-      .catch(console.error);
-  }, [api]);
+  const { value: data } = useFetch<PoolData>(
+    backendUrl[currentChainId] + '/amm/arbitrage',
+    {},
+  );
 
-  useEffect(() => {
-    // Only show component if you can earn > 0.001 BTC or > 1 USDT
-    if (
-      (data.USDT.rateToBalance.earn > 0.001 &&
-        symbolByTokenAddress(data.USDT.rateToBalance.to) === 'RBTC') ||
-      (data.USDT.rateToBalance.earn > 10 &&
-        symbolByTokenAddress(data.USDT.rateToBalance.to) === 'USDT')
-    ) {
-      setShow(true);
+  const opportunityArray = useMemo(() => {
+    const opportunities: {
+      fromToken: Asset;
+      toToken: Asset;
+      fromAmount: number;
+      toAmount: number;
+      earn: number;
+      earnUsd: number;
+    }[] = [];
+    for (let pool in data) {
+      if (
+        data.hasOwnProperty(pool) &&
+        data[pool].hasOwnProperty('rateToBalance')
+      ) {
+        const toToken = assetByTokenAddress(data[pool].rateToBalance.to);
+        const rate = assetRates.find(
+          item => item.source === toToken && item.target === Asset.USDT,
+        );
+        opportunities.push({
+          fromToken: assetByTokenAddress(data[pool].rateToBalance.from),
+          toToken,
+          fromAmount: data[pool].rateToBalance.amount,
+          toAmount: data[pool].rateToBalance.rate,
+          earn: data[pool].rateToBalance.earn,
+          earnUsd: rate
+            ? Number(
+                bignumber(fixNumber(rate.value.rate))
+                  .mul(data[pool].rateToBalance.earn)
+                  .div(rate.value.precision)
+                  .toFixed(18),
+              )
+            : 0,
+        });
+      }
     }
-  }, [data]);
+    return opportunities.sort((a, b) => b.earnUsd - a.earnUsd);
+  }, [data, assetRates]);
 
-  const fromAmount = data.USDT.rateToBalance.amount.toFixed(4);
-  const fromToken = symbolByTokenAddress(data.USDT.rateToBalance.from);
-  const toAmount = data.USDT.rateToBalance.rate.toFixed(2);
-  const toToken = symbolByTokenAddress(data.USDT.rateToBalance.to);
+  const opportunity = useMemo(() => {
+    const items = opportunityArray.filter(
+      item => item.earnUsd > minUsdForOpportunity,
+    );
+    return items.length ? items[0] : null;
+  }, [opportunityArray]);
 
   return (
     <>
-      {show && (
+      {opportunity !== null && (
```

Comment:
The `render` needs to be adjusted for mobile devices, this is what happens under `992px` width:
![image](https://user-images.githubusercontent.com/14878713/120496147-9f132780-c3bd-11eb-8613-54b9a19f6f52.png)

Reply:
Oh, still looks cool :D